package/gluon-config-mode-domain-select/luasrc/lib/gluon/config-mode/wizard/0200-domain-select.lua

```diff
@@ -0,0 +1,59 @@
+return function(form, uci)
+	local site = require 'gluon.site'
+	local path = '/lib/gluon/domains'
+	local fs = require 'nixio.fs'
+
+	local function read_json(path)
+
+		local json = require 'luci.jsonc'
+		local decoder = json.new()
+		local sink = decoder:sink()
+
+		local file = assert(io.open(path))
+
+		while true do
+			local chunk = file:read(2048)
+			if not chunk or chunk:len() == 0 then break end
+			sink(chunk)
+		end
+
+		file:close()
+
+		return assert(decoder:get())
+	end
+
+	local function get_domain_list()
+		local list = {}
+		for domain_path in io.popen('find ' .. path .. '/ -iname \*.json -print'):lines() do
+			local domain_code = domain_path:match(path .. '/(.*)\.json$')
+			table.insert(list, domain_code)
+		end
+		return list
+	end
+
+	local s = form:section(Section, nil, translate('gluon-config-mode:domain-select'))
+
+	local o = s:option(ListValue, 'domain', translate('gluon-config-mode:domain'))
+	o.optional = false
```

Comment:
Please set `o.default` so an already selected domain is preselected in the select box.